### PR TITLE
Fix for minyear and maxyear in the calendar.js

### PR DESCRIPTION
**Pull request by pinta83**
/media/system/js/fields/calendar.js at lines 861 and 864 call
date.getOtherFullYear(this.params.dateType, year);

But this function defined in
/media/system/js/fields/calendar-locales/date/jalali/date-helper.js
does not have a second parameter - year, so it can not be changed...

Here's a fix for this issue...

Pull Request for Issue #19926 .

### Summary of Changes
Changes 2 lines of code so the picker would work as inteded

### Testing Instructions
Create a datepicker input field with attribute maxyear set to -5

### Expected result
When the field gets rendered the expected result of the initially displayed year - is the current year -5 years as defined.

### Actual result
The year title is ok, but the calendar displayed does not reflect the shown year, but the current month is displayed. When you click the previous year button, you have to click 5 times until it gets to the right year, and then starts working as expected. It also lets you pick the date outside of range. This fixes all these issuse, as the wrong calendar month never gets displayed.

### Documentation Changes Required
None

### CODE REVIEW

**Comment by pinta83** [comment]
Have in mind that the minified version (media/system/js/fields/calendar.min.js - NOT PROVIDED) should be updated aswell

**Comment by ghost** [comment]
@pinta83 please fill the Template (Summary of Changes
Testing Instructions
Expected result
Actual result
Documentation Changes Required)

**Comment by pinta83** [comment]
@franz-wohlkoenig Done.. sry ;)

**Comment by Quy** [comment]
Please use the following tool to compile/include the minified version.
https://skalman.github.io/UglifyJS-online/

**Comment by pinta83** [comment]
@Quy Done..

File: media/system/js/fields/calendar.js [hunk 1]
```diff
@@ -858,10 +858,10 @@
 
 		if (year < this.params.minYear) {                                                                   // Check min,max year
 			year = this.params.minYear;
-			date.getOtherFullYear(this.params.dateType, year);
+			date.setYear(year);
 		} else if (year > this.params.maxYear) {
```
Comment (dgrammatiko) [inline media/system/js/fields/calendar.js:862]: This seems wrong, it seems that it should be 
```js
		date.setOtherFullYear(this.params.dateType, year);
```

**Comment by pinta83** [comment]
Fixed, tested and updated

File: media/system/js/fields/calendar.js [hunk 2]
```diff
@@ -858,10 +858,10 @@
 
 		if (year < this.params.minYear) {                                                                   // Check min,max year
 			year = this.params.minYear;
-			date.getOtherFullYear(this.params.dateType, year);
+			date.setOtherFullYear(this.params.dateType, year);
 		} else if (year > this.params.maxYear) {
 			year = this.params.maxYear;
-			date.getOtherFullYear(this.params.dateType, year);
+			date.setOtherFullYear(this.params.dateType, year);;
```
Comment (C-Lodder) [inline media/system/js/fields/calendar.js]: You have a double semi-colon at the end of this

Comment (pinta83) [reply media/system/js/fields/calendar.js]: Fixed

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 922baabe2b3f39af08c96e8d5c690bbd780f0df0<br /><br />Hint for other testers: Instead of creating a field, you can modify field "created" in file "administrator/components/com_contact/models/forms/contact.xml" by adding property `maxyear="-19"` (orn other values). Then create a contact and play with the created date calendar field as described in the testing instructions and the issue.

After having applied the patch e.g. with patchtester, you might have to clear your browser cache in order to use the changed js (or minified js if debug is switched off).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26119">issues.joomla.org/tracker/joomla-cms/26119</a>.</sub>

**Comment by infograf768** [comment]
@dgrammatiko 
can you have a look please?

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 922baabe2b3f39af08c96e8d5c690bbd780f0df0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26119">issues.joomla.org/tracker/joomla-cms/26119</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by HLeithner** [comment]
Thanks for fixing the calendar.

**Pull request merged**